src/app/find/locationDetails/ShelterService.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react';
+import moment from 'moment';
+import PhoneLink from '../../../components/phoneLink';
+import ServiceOfferings from './ServiceOfferings';
+import ServiceRestrictions from './ServiceRestrictions';
+import Icon from '../../../components/icon';
+import { DAYS, OCCASIONS } from '../../../Constants';
+import InfoItem from './InfoItem';
+
+const renderSchedule = (schedule) => {
+  if (schedule.length === 7 &&
+    schedule.every(day => day.opens_at === '00:00:00' && day.closes_at === '23:59:00')) {
+    return 'Open 24/7';
+  }
+
+  const dayNumberToName = weekday => DAYS[weekday - 1];
+
+  const formatHour = time => moment(time, 'HH:mm:ss').format('LT').replace(':00 ', ' ');
+  const formatHours = (opens, closes) => `${formatHour(opens)} to ${formatHour(closes)}`;
+
+  const formatRange = ({ start, end }) => {
+    if (end === start) {
+      return dayNumberToName(start);
+    }
+    if (end === start + 1) {
+      return `${dayNumberToName(start)} & ${dayNumberToName(end)}`;
+    }
+    return `${dayNumberToName(start)} to ${dayNumberToName(end)}`;
+  };
+
+  const orderedSchedule = schedule.sort(({ weekday: day1 }, { weekday: day2 }) => day1 - day2);
+
+  const daysGroupedByHours = orderedSchedule.reduce((grouped, day) => {
+    const hoursString = formatHours(day.opens_at, day.closes_at);
+    return { ...grouped, [hoursString]: [...(grouped[hoursString] || []), day.weekday] };
+  }, {});
+
+  const groupStrings = Object.keys(daysGroupedByHours).map((hoursString) => {
+    const dayRanges = [];
+    const remainingWeekdays = daysGroupedByHours[hoursString];
+
+    while (remainingWeekdays.length) {
+      const currentDayRange = dayRanges[dayRanges.length - 1];
+      const day = remainingWeekdays.shift();
+
+      if (currentDayRange && currentDayRange.end === day - 1) {
+        currentDayRange.end = day;
+      } else {
+        dayRanges.push({ start: day, end: day });
+      }
+    }
+
+    return `${dayRanges.map(formatRange).join(', ')} ${hoursString}`;
+  });
+
+  return `Open ${groupStrings.join(', ')}`;
+};
+
+function ShelterService({ service, selectedCategory, showOnMount }) {
+  const [isShown, setIsShown] = useState(showOnMount);
```

Comment:
Nitpick, but maybe all these instances of "shown" would be more accurate as "expanded".
I mean, the service is _shown_ either way - just the header instead of the full contents when it's collapsed.